src/renderers/WebGLRenderer.js

```diff
@@ -206,6 +211,8 @@ function WebGLRenderer( parameters ) {
 
 		}
 
+		_gl.isWebGL2 = typeof WebGL2RenderingContext !== 'undefined' && _gl instanceof WebGL2RenderingContext;
```

Comment:
I wouldn't pollute the webgl context. What about adding a `isWebGL2` attribute on the WebGLRenderer itself?

Reply:
I think the problem is that certain modules of the renderer like `WebGLBufferRenderer` do not have a reference to the renderer but to the raw WebGL context instead. If we add `isWebGL2` to `WebGLRenderer`, we need to pass it to the respective modules.

Reply:
Yeah I was thinking about it too, in any case the change will be small, just replacing the WebGLcontext parameter by WebGLRenderer and access the context from the renderer inside the module. Not strong opinion either, but If we pollute the webgl context, should we open the door to practices like babylon's https://github.com/mrdoob/three.js/pull/13717#discussion_r178969902 ?
I guess @mrdoob should decide here :)

Reply:
Yup, I was gonna comment the same thing about no reference to `WebGLRenderer` from some modules. I don't prefer passing `WebGLRenderer` instance to such modules instead of `WebGLContext` because I don't wanna let them have dependency with `WebGLRenderer`.

And also, tbh I don't really wanna pollute `WebGLContext` object. So... `.isWebGL2( gl )` function in utils.js? Not strong preference tho.

Reply:
Yes, I think I would add a `isWebGL2()` function to `WebGLUtils`. Do you mind doing a PR with the change?